### PR TITLE
feat : add more flexibility to the client cli

### PR DESCRIPTION
**Pull request by leunam217**
The print-server and print-token commands don't ask you to enter information if there is not a config file.

### CODE REVIEW

**Comment by leunam217** [comment]
As suggested in https://github.com/ocaml-sf/learn-ocaml/issues/279#issuecomment-508755531 I introduce a "init" command

File: src/main/learnocaml_client.ml [hunk 1]
```diff
@@ -609,7 +609,7 @@ let get_config ?local ?(save_back=false) server_opt token_opt =
   get_config_option ?local ~save_back server_opt token_opt
   >>= function
   | Some c -> Lwt.return c
-  | None -> init ?local ?server:server_opt ?token:token_opt ()
+  | None -> Lwt.fail_with "Not config file found please do learn-ocaml-client init"
```
Comment (yurug) [inline src/main/learnocaml_client.ml]:
```suggestion
  | None -> Lwt.fail_with "No config file found. Please do `learn-ocaml-client init`"
```

File: src/main/learnocaml_client.ml [hunk 2]
```diff
@@ -628,6 +628,47 @@ let get_config_o ?save_back o =
   let open Args_global in
   get_config ~local:o.local ?save_back o.server_url o.token
 
+module Init = struct
+  open Args_global
+  open Args_create_token
+     
+  let init global_args create_token_args =
+  let path = if global_args.local then ConfigFile.local_path else ConfigFile.user_path in
```
Comment (yurug) [inline src/main/learnocaml_client.ml]: There is an indentation issue here.

File: src/main/learnocaml_client.ml [hunk 3]
```diff
@@ -628,6 +628,47 @@ let get_config_o ?save_back o =
...
+  let get_token server =
+    match global_args.token with
+    | Some token -> Lwt.return token
+    | None ->
+       match create_token_args.nickname with
+       | Some n -> (get_nonce_and_create_token server (Some n) create_token_args.secret)
+       | _ -> Lwt.fail_with "You must provide a token or a nickname and a secret"
```
Comment (yurug) [inline src/main/learnocaml_client.ml]:
```suggestion
       | _ -> Lwt.fail_with "You must provide a token or a nickname and a secret."
```

File: src/main/learnocaml_client.ml [hunk 4]
```diff
@@ -628,6 +628,47 @@ let get_config_o ?save_back o =
...
+  in
+  let get_server () =
+    match global_args.server_url with
+    | None -> Lwt.fail_with "You must provide a server"
```
Comment (yurug) [inline src/main/learnocaml_client.ml]:
```suggestion
    | None -> Lwt.fail_with "You must provide a server."
```

File: src/main/learnocaml_client.ml [hunk 5]
```diff
@@ -628,6 +628,47 @@ let get_config_o ?save_back o =
...
+    | Some s -> Lwt.return s
+  in
+  get_server () >>= fun server ->
+  check_server_version server >>= fun () ->
+  get_token server >>= fun token ->
+  let config = { ConfigFile. server; token } in
+  ConfigFile.write path config >|= fun () ->
+  Printf.eprintf "Configuration written to %s\n%!" path;
```
Comment (yurug) [inline src/main/learnocaml_client.ml]:
```suggestion
  Printf.eprintf "Configuration written to %s.\n%!" path;
```

File: src/main/learnocaml_client.ml [hunk 6]
```diff
@@ -628,6 +628,47 @@ let get_config_o ?save_back o =
...
+  0
+
+  let man = man "Initialize the configuration file with the server and : \
```
Comment (yurug) [inline src/main/learnocaml_client.ml]:
```suggestion
  let man = man "Initialize the configuration file with the server, and \
```

**Comment by yurug** [comment]
Thanks!

**Comment by yurug** [comment]
There is still a failed test in the CI. CC @leunam217

**Comment by leunam217** [comment]
Normally now it's ok @yurug

**Comment by yurug** [comment]
Thanks!

**Pull request merged**